engine/src/main/java/org/terasology/world/block/family/UpdatesWithNeighboursFamilyFactory.java

```diff
@@ -84,74 +87,66 @@ protected UpdatesWithNeighboursFamilyFactory(ConnectionCondition connectionCondi
+        basicBlocks[3].put(shapeMapping.get(THREE_CONNECTIONS_CORNER), THREE_CONNECTIONS_CORNER);
+        basicBlocks[3].put(shapeMapping.get(THREE_CONNECTIONS_T), THREE_CONNECTIONS_T);
 
         basicBlocks[4] = new TByteObjectHashMap<>();
-        putBlockDefinition(basicBlocks[4], blockBuilder, blockDefJson, FOUR_CONNECTIONS_CROSS);
-        putBlockDefinition(basicBlocks[4], blockBuilder, blockDefJson, FOUR_CONNECTIONS_SIDE);
+        basicBlocks[4].put(shapeMapping.get(FOUR_CONNECTIONS_CROSS), FOUR_CONNECTIONS_CROSS);
+        basicBlocks[4].put(shapeMapping.get(FOUR_CONNECTIONS_SIDE), FOUR_CONNECTIONS_SIDE);
 
         basicBlocks[5] = new TByteObjectHashMap<>();
-        putBlockDefinition(basicBlocks[5], blockBuilder, blockDefJson, FIVE_CONNECTIONS);
+        basicBlocks[5].put(shapeMapping.get(FIVE_CONNECTIONS), FIVE_CONNECTIONS);
 
         basicBlocks[6] = new TByteObjectHashMap<>();
-        putBlockDefinition(basicBlocks[6], blockBuilder, blockDefJson, SIX_CONNECTIONS);
+        basicBlocks[6].put(shapeMapping.get(SIX_CONNECTIONS), SIX_CONNECTIONS);
 
-        BlockUri blockUri = new BlockUri(blockDefUri.getModuleName(), blockDefUri.getAssetName());
+        BlockUri blockUri = new BlockUri(definition.getUrn());
 
         // Now make sure we have all combinations based on the basic set (above) and rotations
         for (byte connections = 0; connections < 64; connections++) {
             // Only the allowed connections should be created
             if ((connections & connectionSides) == connections) {
-                Block block = constructBlockForConnections(connections, blockBuilder, blockDefUri, basicBlocks);
+                Block block = constructBlockForConnections(connections, blockBuilder, definition, basicBlocks);
                 if (block == null) {
                     throw new IllegalStateException("Unable to find correct block definition for connections: " + connections);
                 }
-                block.setUri(new BlockUri(blockUri, String.valueOf(connections)));
+                block.setUri(new BlockUri(blockUri, new Name(String.valueOf(connections))));
                 blocksForConnections.put(connections, block);
             }
         }
 
         final Block archetypeBlock = blocksForConnections.get(SideBitFlag.getSides(Side.RIGHT, Side.LEFT));
-        return new UpdatesWithNeighboursFamily(connectionCondition, blockUri, blockDefinition.categories,
+        return new UpdatesWithNeighboursFamily(connectionCondition, blockUri, definition.getCategories(),
                 archetypeBlock, blocksForConnections, connectionSides);
     }
 
-    private void putBlockDefinition(TByteObjectMap<BlockDefinition> blockDefinitions, BlockBuilderHelper blockBuilder, JsonObject blockDefJson,
-                                    String connections) {
-        Byte value = shapeMapping.get(connections);
-        if (value != null) {
```

Comment:
Is there a reason why you aren't checking anymore for null? e.g. must users provide values for all keys?

The Fences module FencesFamilyFactory doesn't pass all keys in FencesFamilyFactory